lib/elasticsearch/bulk.js

```diff
@@ -98,46 +128,41 @@ Bulk.prototype._fn = function(d) {
         return;
 
       // Insert a copy of the original body
-      e.items.forEach((e,i) =>e.body = d[i]);
+      e.items.forEach((e,i) => e.body = itemsToProcess[i * 2 + 1]);
+
+      if (this.options.pushResults) {
+        if (this.options.maxRetries) {
+          let itemsToRetry;
+          e.items.forEach((item, index) => {
+            const verb = item.update || item.index || item.create;
+            if (verb && verb.error) {
+              itemsToRetry = itemsToRetry || [];
+              itemsToRetry.push(itemsToProcess[index * 2]);
+              itemsToRetry.push(itemsToProcess[index * 2 + 1]);
+            }
+            else {
+              itemsSuccessfullyPushed.push(item);
+            }
+          });
+
+          if (itemsToRetry) {
+            itemsToProcess = itemsToRetry;
```

Comment:
I wonder if we should make `itemToProcess` an input to execute, might be easier to follow, as this would be `return execute(itemsToRetry)`